services/steam/steam.service.js

```diff
@@ -0,0 +1,395 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+const { metric, formatDate } = require('../../lib/text-formatters')
+const { age: ageColor } = require('../../lib/color-formatters')
+const prettyBytes = require('pretty-bytes')
+
+const steamCollectionSchema = Joi.object({
+  response: Joi.object()
+    .keys({
+      result: Joi.number()
+        .integer()
+        .min(1)
+        .max(1)
+        .required(),
+      resultcount: Joi.number()
+        .integer()
+        .min(0)
+        .max(1)
+        .required(),
+      collectiondetails: Joi.array()
+        .items(
+          Joi.object({
+            publishedfileid: Joi.string(),
+            result: Joi.number().integer(),
+            children: Joi.array(),
+          })
+        )
+        .required(),
+    })
+    .required(),
+}).required()
+
+const steamFileSchema = Joi.object({
+  response: Joi.object()
+    .keys({
+      result: Joi.number()
+        .integer()
+        .min(1)
+        .required(),
+      resultcount: Joi.number()
+        .integer()
+        .min(0)
+        .required(),
+      publishedfiledetails: Joi.array().items(
+        Joi.object({
+          publishedfileid: Joi.string().required(),
+          result: Joi.number()
+            .integer()
+            .required(),
+          file_size: Joi.number().integer(),
+          time_created: Joi.number().integer(),
+          subscriptions: Joi.number().integer(),
+          favorited: Joi.number().integer(),
+          lifetime_subscriptions: Joi.number().integer(),
+          lifetime_favorited: Joi.number().integer(),
+          views: Joi.number().integer(),
```

Comment:
Again, I think all of these properties should be `.required()`. We need to specify `.required()` in order to throw an error if it isn't present in the response.

Reply:
If the item is not found is results in a JSON format looking like this:
```json
{"response":{"result":1,"resultcount":1,"publishedfiledetails":[{"publishedfileid":"-1","result":9}]}}
```

With the request being successful `response.result == 1` but the item not being found `response.publishedfiledetails.result == 9`. 

If I say these things need to be required the schema fails right? Are there examples of how to use this to catch errors?